### DISCUSSION TITLE
How to write a function to `clone` & then `cd` into the local repository?

### DISCUSSION THREAD

**Discussion by jarmos-san** [fish-shell/fish-shell#8975] [Q&A] [answered]
I'm trying to create a function to clone a `git` remote repository on to my local machine & then `cd` into the local repository. So, any ideas how can I achieve what am I trying to do?

On Bash, the following snippet worked great for me ([source](https://unix.stackexchange.com/a/702172)):
```bash
 gclone () {
    git clone $1
    local ldir=`ls -t | head -1`
    echo $ldir
    cd $ldir
    pwd
 }
```

**Comment by faho** [comment, marked as answer]
Please look at the documentation, in particular https://fishshell.com/docs/current/fish_for_bash_users.html.

In your case, this is simple enough to rewrite directly:

```fish
function gclone
    git clone $argv[1]
    set -l ldir (ls -t | head -1)
    echo $ldir
    cd $ldir
    pwd
end
```

However, it would scale a lot better if you read the documentation for yourself before you asked.
